pkg/asset/agent/configimage/configiso.go

```diff
@@ -0,0 +1,159 @@
+	"github.com/openshift/installer/pkg/asset/agent/image"
+)
+
+const (
+	configImageFilename = "agentconfig.iso"
+
+	pullSecretManifest          string = "/etc/assisted/manifests/pull-secret.yaml" //nolint:gosec // not hardcoded credentials
+	nmStateConfigManifest       string = "/etc/assisted/manifests/nmstateconfig.yaml"
+	clusterDeploymentManifest   string = "/etc/assisted/manifests/cluster-deployment.yaml"
+	clusterImageSetManifest     string = "/etc/assisted/manifests/cluster-image-set.yaml"
+	agentClusterInstallManifest string = "/etc/assisted/manifests/agent-cluster-install.yaml"
+	extraManifests              string = "/etc/assisted/extra-manifests"
+	hostConfigFiles             string = "/etc/assisted/hostconfig"
+	hostNameFiles               string = "/etc/assisted/hostnames"
+	kubeadminPasswordHash       string = "/opt/agent/tls/kubeadmin-password.hash" //nolint:gosec // not hardcoded credentials
+	kubeconfigTLSCerts          string = "/opt/agent/tls/admin-kubeconfig-signer"
+	kubeapiserverTLSCerts       string = "/opt/agent/tls/kube-apiserver"
+)
+
+// TODO - need to add rendezvous host file when available.
+var fileFilters = []string{
+	pullSecretManifest,
+	nmStateConfigManifest,
+	clusterDeploymentManifest,
+	clusterImageSetManifest,
+	agentClusterInstallManifest,
+	extraManifests,
+	hostConfigFiles,
+	hostNameFiles,
+	kubeadminPasswordHash,
+	kubeconfigTLSCerts,
+	kubeapiserverTLSCerts,
+}
+
+// ConfigImage is an asset that generates a configuration ISO that can be used configure hosts.
+type ConfigImage struct {
+	tmpPath string
+}
+
+var _ asset.WritableAsset = (*ConfigImage)(nil)
+
+// Dependencies returns the assets on which the Config asset depends.
+func (a *ConfigImage) Dependencies() []asset.Asset {
+	return []asset.Asset{
+		&image.Ignition{},
+	}
+}
+
+// Generate generates the configuration image file.
+func (a *ConfigImage) Generate(dependencies asset.Parents) error {
+	ignition := &image.Ignition{}
+
+	dependencies.Get(ignition)
+
+	ca := image.NewCpioArchive()
+
+	// Walk the list of ignition files.
+	// For the ones to be included in the configimage add to cpio archive
+	for _, file := range ignition.Config.Storage.Files {
+		filename := file.Node.Path
+		mode := *file.FileEmbedded1.Mode
+		if includeFileInImage(filename) {
+			logrus.Debugf("Including file %s", filename)
+			contents := ""
+			if file.FileEmbedded1.Contents.Source != nil {
+				output, err := dataurl.DecodeString(*file.FileEmbedded1.Contents.Source)
+				if err != nil {
+					return errors.Wrapf(err, "decode string failed")
+				}
+				contents = string(output.Data)
+			}
+
+			err := ca.StoreBytes(filename, []byte(contents), mode)
+			if err != nil {
+				return errors.Wrapf(err, "failure storing file in cpio")
+			}
+		}
+	}
+
+	// Save the cpi archive to a temp file
+	tmpPath, err := os.MkdirTemp("", "agent")
+	if err != nil {
+		return err
+	}
+	a.tmpPath = tmpPath
+
+	output := filepath.Join(tmpPath, "configiso.gz")
```

Comment:
This name seems to be slightly misleading, this is a compressed cpio archive, not an iso?

Reply:
Yes I will fix that name.